skeletonkey/config.py

```diff
@@ -346,17 +346,31 @@ def add_args_from_dict(
                 if key[1:] in os.environ:
                     value = os.environ[key[1:]]
                 arg_parser.add_argument(
-                    f"--{prefix}{key[1:]}", default=value, type=type(value)
+                    f"--{prefix}{key[1:]}", default=value
                 )
             elif key.startswith("?"):
                 arg_parser.add_argument(
                     f"--{prefix}{key[1:]}", default=value, action='store_true'
                 )
             else:
                 arg_parser.add_argument(
-                    f"--{prefix}{key}", default=value, type=type(value)
+                    f"--{prefix}{key}", default=value
                 )
 
+def update_flat_config_types(flat_config: Config) -> Config:
+    """
+    Given a flat config containing some string values, parse those string values as if they were
+    yaml arguemnts into the corresponding python type and return an updated config.
+
+    Args:
+        config (Config): The flat Config whose values should be parsed
+    """
+    return Config({

```

Comment:
Please add a rule so that a user can't write out a yaml formatted string from the command line. We only want this to convert a single string into its correct type, not many key value pairs.